.github/actions/regression-tests/action.yml

```diff
@@ -47,8 +49,23 @@ runs:
         # used by PyTests
         export DRAGONFLY_PATH="${GITHUB_WORKSPACE}/${{inputs.build-folder-name}}/${{inputs.dfly-executable}}"
 
-        pytest -m "${{inputs.filter}}" --json-report --json-report-file=rep1_report.json dragonfly/replication_test.py --df alsologtostderr --df enable_multi_shard_sync=true
-        pytest -m "${{inputs.filter}}" --json-report --json-report-file=rep2_report.json dragonfly/replication_test.py --df alsologtostderr --df enable_multi_shard_sync=false
+        timeout 20m pytest -m "${{inputs.filter}}" --json-report --json-report-file=rep1_report.json dragonfly/replication_test.py --log-cli-level=INFO --df alsologtostderr --df enable_multi_shard_sync=true || code=$?; if [[ $code -eq 124 ]]; then echo "TIMEDOUT=1">> "$GITHUB_OUTPUT"; exit 1; fi
+
+
+        timeout 20m pytest -m "${{inputs.filter}}" --json-report --json-report-file=rep2_report.json dragonfly/replication_test.py --log-cli-level=INFO --df alsologtostderr --df enable_multi_shard_sync=false || code=$?; if [[ $code -eq 124 ]]; then echo "TIMEDOUT=1">> "$GITHUB_OUTPUT"; exit 1; fi
+
+    - name: Print last log on timeout
+      if: failure()
+      shell: bash
+      env:
+        TIMEDOUT_STEP_1: ${{ steps.first.outputs.TIMEDOUT }}
+        TIMEDOUT_STEP_2: ${{ steps.second.outputs.TIMEDOUT }}
+      run: |
+        if [[ "${{ env.TIMEDOUT_STEP_1 }}" -eq 1 ]] || [[ "${{ env.TIMEDOUT_STEP_2 }}" -eq 1 ]]; then
+          echo "🪵🪵🪵🪵🪵🪵 Latest log before timeout 🪵🪵🪵🪵🪵🪵\n\n"
+          ${GITHUB_WORKSPACE}/tools/extract_latest_log.py --path /tmp/ | xargs cat
+          echo "🪵🪵🪵🪵🪵🪵 Latest log before timeout end 🪵🪵🪵🪵🪵🪵\n\n"
```

Comment:
I could have just printed the file on the python script but honestly I should also have used the bash itself -- really the python script was an overkill -- so I removed it :smile: 

Yes I needed the `xargs cat` because otherwise only the filename would get printed and not its contents

+1 it was over-engineered for no reason

Reply:
Ah yes, I forgot the script only prints the name 😅